In [2]:
import numpy as np
import scipy.stats as stats
import statsmodels
import warnings
warnings.filterwarnings('ignore')
from statsmodels.stats.weightstats import ztest

In [3]:
### Given data
n=25
x_bar=3050
pop_mean=mu=1000+3000
sigma=5*25
#alpha=0.05 CI=0.975

In [4]:
#2.Calculate the Test Statistic:
zscore=(x_bar-pop_mean)/(sigma/np.sqrt(n))
zscore

np.float64(-38.0)

In [5]:
## Ztable value
ztable=stats.norm.ppf(0.975)
ztable

np.float64(1.959963984540054)

In [6]:
p_value = 1 - stats.norm.cdf(zscore)
p_value

np.float64(1.0)

In [7]:
p=1-stats.norm.cdf(25,loc=pop_mean,scale=125)
p

np.float64(1.0)

# ESTIMATION AND CONFIDENCE INTERVALS

In [16]:
# Given Data
x=[1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
n=len(x)
x_bar=np.mean(x)
x_bar
s=np.std(x)


In [28]:
## t critical values
t_critical = stats.t.ppf(1 - alpha/2, df)
t_critical

np.float64(2.976842734370834)

In [27]:
confidence = 0.99
alpha = 1 - confidence
df = n - 1
t_critical = stats.t.ppf(1 - alpha/2, df)
margin_error = t_critical * (s / np.sqrt(n))
lower = x_bar - margin_error
upper =x_bar+ margin_error
print("99% CI using Sample SD (t-distribution):")
print(lower, upper)

99% CI using Sample SD (t-distribution):
1.095231668626123 1.3821016647072102


In [29]:
### Margin Error
Margin_error=t_critical  *(s/np.sqrt(n))
Margin_error

np.float64(0.14343499804054372)

Interpretation:
We are 99% confident that the true mean durability of the print-heads lies between1.095 and 1.382 million characters before failure.

In [ ]:
When the population variance (or standard deviation) is known, we use the Z-test

In [20]:
# Given data
sig=0.2

In [24]:
sigma = 0.2
z_critical = stats.norm.ppf(1 - alpha/2)
margin_error_z = z_critical * (sigma / np.sqrt(n))
lower_z = x_bar - margin_error_z
upper_z = x_bar+ margin_error_z
print("99% CI using Population SD (Z-distribution):")
print(lower_z, upper_z)

99% CI using Population SD (Z-distribution):
1.1056514133957607 1.3716819199375725


In [ ]:
Interpretation:
We are 99% confident that the true mean durability of the print-heads lies between 1.105 and 1.371 million characters before failure.